src/Interpreters/Aggregator.cpp

```diff
@@ -1193,6 +1193,9 @@ bool Aggregator::checkLimits(size_t result_size, bool & no_more_keys) const
         }
     }
 
+    /// Some aggregate functions cannot throw exceptions on allocations (e.g. from C malloc)
```

Comment:
Is it ok to just return a `nullptr` when memory limit is exceeded?

Reply:
Generally I won't do it, many libraries are not ready (they will either segfault on nullptr dereference or abort).
Maybe Roaring Bitmap is ok but I don't want to "surprise" it.

E.g. I expect it may have memory leak if one of subsequent allocations return nullptr.

Reply:
PS. You might noticed, we have removed throwing from operator new.